paper.md

```diff
@@ -2,11 +2,88 @@
+
+<!-- this is what the above problems don't solve -->
+
+After studying the different workflow execution engines available, we discovered
+the absence of a possible third category, which should enforce the execution of workflows inside containers, hence making it container-native but should not enforce the presence of a Kubernetes cluster or a cloud computing environment. The presence of a Kubernetes cluster or a cloud computing environment should not be a hard core requirement for reproducing any experiment since it is often costly [@rodriguez2020container] to get access to one and inturn makes reproducibilty complex. Hence, it should provide flexibility for their users to run workflows in a wide range of computing environments like a single node local machine, a HPC environment [@yang2005high], a Kubernetes cluster or any cloud computing environment of their choice.
+
+<!-- this is our contributions -->
+
+Through this paper, we introduce a tool called Popper, which follows a container-native strategy for building reproducible workflows from archived experimental artifacts [@jimenez_popper_2017]. The core idea behind this tool is that it breaks a complex workflow into discrete steps and each step
+executes inside a separate container. These containers can run on a wide variety of computing environments like the local machine, the cloud or an HPC environment [@containerbasedvirt2013]. We provide a detailed description of the internals of the tool in section II. In section III, we present some case studies of how Popper can be used to quickly reproduce complex workflows in different computing environments. We show how an entire Machine Learning workflow can be run on a local machine during development and how it can be reproduced in a Kubernetes cluster with GPU's to scale up and collect results. We also show how a HPC workflow developed on the local machine can be reproduced easily in a SLURM cluster [@slurm]. Then we talk about how Popper differs from existing workflow execution engines in section V. We conclude with discussing some benefits and challenges of using Popper for solving the reproducibilty problem.
```

Comment:
let's add these contributions in the form of a list. "The contributions of this paper are: <list>"